### PR TITLE
Adding db replica datasource #489

### PR DESCRIPTION
**Pull request by danaelhe**
running into a strange "id" attribute issue when testing it

### CODE REVIEW

File: digitalocean/datasource_digitalocean_database_replica.go
Comment (andrewsomething) [inline digitalocean/datasource_digitalocean_database_replica.go:108]:
```suggestion
	clusterId :=  d.Get("cluster_id").(string)
	name := d.Get("name").(string)
	d.Set("cluster_id", clusterId)
	d.Set("name", name)
	d.SetId(makeReplicaId(clusterId, name))
```

You'll need to set a unique ID for the data source. We already have a little helper to make one from the cluster ID and replica name:

https://github.com/digitalocean/terraform-provider-digitalocean/blob/c73e7a5cf4fbe88060c471a808acf862c327b893/digitalocean/resource_digitalocean_database_replica.go#L208-L210

File: digitalocean/datasource_digitalocean_database_replica_test.go
Comment (andrewsomething) [inline digitalocean/datasource_digitalocean_database_replica_test.go:72]:
```suggestion
					resource.TestCheckResourceAttrPair("digitalocean_database_replica.read-01", "host",
						"data.digitalocean_database_replica.my_db_replica", "host"),
								),
```

Would be great to test that the computed attributes are set as expected. In fact, it's more important to test that here than in the step above which is really just setting things up for this test.

File: digitalocean/datasource_digitalocean_database_replica_test.go
Comment (andrewsomething) [inline digitalocean/datasource_digitalocean_database_replica_test.go:71]:
```suggestion
					resource.TestCheckResourceAttrPair("digitalocean_database_replica.read-01", "cluster_id",
						"data.digitalocean_database_replica.my_db_replica", "cluster_id"),
					resource.TestCheckResourceAttrPair("digitalocean_database_replica.read-01", "name",
						"data.digitalocean_database_replica.my_db_replica", "name"),
```

To reference an attribute of a data source, you needed to specify `data.<type>.<name>`

File: digitalocean/datasource_digitalocean_database_replica.go
Comment (scotchneat) [reply digitalocean/datasource_digitalocean_database_replica.go:108]: Ah nice.

File: digitalocean/datasource_digitalocean_database_replica_test.go
Comment (scotchneat) [reply digitalocean/datasource_digitalocean_database_replica_test.go:72]: We were troubleshooting the id issue but this was next

File: digitalocean/datasource_digitalocean_database_replica.go
Comment (danaelhe) [reply digitalocean/datasource_digitalocean_database_replica.go]: thanks andrew!

**Comment by scotchneat** [comment]
Looks good @danaelhe!
Once you merge the latest changes, I'll approve.

**Comment by danaelhe** [comment]
> Looks good @danaelhe!
> Once you merge the latest changes, I'll approve.

hmm. It appears waiting 10 seconds won't cut it anymore. Waiting 10 seconds was consistently passing the other day. Will have to run a few more tests.

File: digitalocean/datasource_digitalocean_database_replica_test.go
Comment (danaelhe) [reply digitalocean/datasource_digitalocean_database_replica_test.go:63]: Thanks! Will make a note to reach out to DBaaS.

File: digitalocean/datasource_digitalocean_database_replica.go
Comment (danaelhe) [reply digitalocean/datasource_digitalocean_database_replica.go:18]: Ah i see. I've removed them from all attributes.

File: digitalocean/datasource_digitalocean_database_replica_test.go
Comment (andrewsomething) [inline digitalocean/datasource_digitalocean_database_replica_test.go:49]: Is the sleep needed here? Would have thought that once the replica is successfully created, the data source should be able to reach it without the sleep.

**Review by andrewsomething — APPROVED** [review]
:+1:  This looks good! 

A couple more comments inline. We'll also need a page for the docs. Here's an example for a data source:

https://github.com/digitalocean/terraform-provider-digitalocean/blob/main/docs/data-sources/database_cluster.md

You can probably reuse most of the descriptions from the resource: https://github.com/digitalocean/terraform-provider-digitalocean/blob/main/docs/resources/database_replica.md

This can be a fast follow if you'd like to get this merged.

File: digitalocean/datasource_digitalocean_database_replica_test.go
Comment (danaelhe) [reply digitalocean/datasource_digitalocean_database_replica_test.go:49]: Oh I had that in for testing and was meant to remove it. good catch!

**Comment by danaelhe** [comment]
@andrewsomething I took a crack at writing out the documentations and I had a few questions:
I only put cluster_id and name as required, and kept tags as optional. I figured the rest (as compared to the resource database replica doc) would be computed?

I'm not 100% sure about the import part either.

File: docs/data-sources/database_replica.md
Comment (andrewsomething) [inline docs/data-sources/database_replica.md:35]: Generally the use case for data sources is to access information about a resource that is not managed in Terraform. So rather than creating the cluster and replica here, a more real world example would probably just have have the data sources perhaps outputting some connection info. 

```suggestion
data "digitalocean_database_cluster" "example" {
  name = "example-cluster"
}

data "digitalocean_database_replica" "read-only" {
  cluster_id = data.digitalocean_database_cluster.example.id
  name       = "terra-test-ro"
}

output "replica_output" {
  value = data.digitalocean_database_replica.read-only.uri
}
```

**Review by andrewsomething — COMMENTED** [review]
> I only put cluster_id and name as required, and kept tags as optional. I figured the rest (as compared to the resource database replica doc) would be computed?

`tags` can also be computed. The only user input is `cluster_id` and `name`.

> I'm not 100% sure about the import part either.

You got it. Importing is for resources, not data sources. Importing brings a resource created outside of Terraform (e.g. the control panel) under management by Terraform. Data sources are inherently read-only, so there is no need for importing them.

https://www.terraform.io/docs/cli/import/index.html

File: docs/data-sources/database_replica.md
Comment (danaelhe) [reply docs/data-sources/database_replica.md:35]: ah yes. that makes sense! thanks.

**Review by ChiefMateStarbuck — COMMENTED** [review]
Looks good! lets give @andrew the final say ;)

**Review by andrewsomething — APPROVED** [review]
LGTM :tada:

**Pull request merged**
